lib/hotel_block.rb

```diff
@@ -0,0 +1,35 @@
+require 'date'
+require_relative 'room'
+require_relative 'reservations'
+
+module Hotel
+    class HotelBlock
+        attr_accessor :date_in, :date_out, :number_of_rooms, :rooms_block
+
+        @@increment_id = 1
+      def initialize(date_in:,date_out:,number_of_rooms:, rooms_block:[], discounted_room_rate:)
+        @id = @@increment_id 
+        @@increment_id += 1
+        @date_in = date_in
+        @date_out = date_out
+        @number_of_rooms = number_of_rooms
+        # this range does not work, its an array of numbers
+        #raise argument error unless (2..5).include? number_of_rooms 
+        @rooms_block =[]
+      end
```

Comment:
I always get suspicious if a class ends up with just `initialize`. It usually means that the class in question is having its methods 'stolen' by other classes.